bundles/binding/org.openhab.binding.zibase/src/main/java/org/openhab/binding/zibase/internal/ZibaseListener.java

```diff
@@ -153,6 +153,12 @@ public void run() {
                     ZbResponse zbResponse = new ZbResponse(receivePacket.getData());
                     logger.debug("ZIBASE MESSAGE: " + zbResponse.getMessage());
                     publishEvents(zbResponse);
+
+                    // reset buffer
```

Comment:
This is really the wrong way to do this.

What should be done is to use the length of the received data to create a zbResponse from the correct set of bytes.

Reply:
Thanks a lot for your prompt review!

Unfortunately the length of the received data is not known. `receivePacket` is filled in by the socket receive, and as it is not clean, there is no way to know the number of bytes received.

Reply:
After investigation, zbReponse expects a byte array of 470 bytes. So I could create one each time:
>                   byte[] readData = new byte[470];
>                   System.arraycopy(receivePacket.getData(), 0, readData, 0, receivePacket.getLength());
>                   ZbResponse zbResponse = new ZbResponse(readData);

However, I don't think it is that more efficient than the reset buffer approach as it allocates a new byte array for each new message...

What would you advise?

Reply:
I'm ok with your Arrays.fill optimization. Not sure if it'd be more efficient to just write all 470 bytes instead of checking the length; probably not a lot of difference either way.